### DISCUSSION TITLE
Azure VM

### DISCUSSION THREAD

**Discussion by ghandlin** [datalust/seq-tickets#1554] [Help]
I have a new install on an Azure VM and having difficulty using a custom url.

I can tracert to the FQDN and it completes in two hops. I have the inbound security rules set up.

![image](https://user-images.githubusercontent.com/1826968/163431715-a8073126-70cb-40b5-b975-a9aaccd7f24f.png)

I'm unsure what else to do.

**Comment by nblumhardt** [comment]
Hi @ghandlin, thanks for your message.

When you say that you're having difficulty with a custom URL, are you able to reach the Seq UI via the default URL or IP address?

It would be great to get an idea of how things fail when you try to reach Seq: is there an HTTP status code/page returned, does the request time out, etc.?

Things to check that are sometimes overlooked:

 * Is the firewall open on the VM? Windows firewall will block 80 and 5341 by default.
 * Is Seq configured with non-default `api.listenUris`? If it's listening at those ports on "localhost" all should be well, but if it's expecting a different hostname, or if the listen URIs use a different scheme such as HTTPS, things won't line up.
 * Is the Seq service running? (`seq service status`)
 * Are there any errors in the most recent logs under `C:\ProgramData\Seq\Logs`?

HTH!
Nick

**Reply by ghandlin** [reply]
@nblumhardt I will check all that and provide more information Monday. However, the browser would just timeout. I did not do anything with the firewall on the VM, just the network security group.

**Reply by nblumhardt** [reply]
Firewall will be blocking by default :+1:

**Reply by ghandlin** [reply]
Ok, so I set up a firewall rule to allow inbound connections on 80, 443 and 5341, but I am still getting a connection timeout. api.listenUris is set to http://[mydomain].eastus.cloudapp.azure.com:5341. I haven't begun setting up https yet, I just want the connection to succeed. I am trying to connect to the custom domain on the local machine. When I have the listener set to http://localhost:5341 all is well.

I checked the log and there was an error about the url not being registered. The service would not start having the custom domain for the listener. I ran the netsh command in the log and the service started, but I am still unable to connect.

**Reply by nblumhardt** [reply]
Thanks for the follow-up. Connecting via the custom domain from the local machine often won't work, depending on network setup.

With `api.listenUris` set to `http://[mydomain].eastus.cloudapp.azure.com:5341`, can you connect to `http://[mydomain].eastus.cloudapp.azure.com:5341` from outside of the Azure network, now that the firewall is open?

**Reply by ghandlin** [reply]
I still cannot connect inside or outside of Azure. 

![image](https://user-images.githubusercontent.com/1826968/163986559-66d18cb0-c791-459d-a630-159a47562df7.png)

![image](https://user-images.githubusercontent.com/1826968/163986705-be012f82-f929-462d-93d0-479063a85c84.png)

**Reply by ghandlin** [reply]
And this is what I had followed prior to working on the firewall on the machine.

https://docs.microsoft.com/en-us/azure/virtual-machines/windows/nsg-quickstart-portal#:~:text=How%20to%20open%20ports%20to%20a%20virtual%20machine,information.%20...%206%20Next%20steps.%20...%20See%20More.

**Reply by ghandlin** [reply]
And @nblumhardt, I really appreciate all the help!

**Reply by nblumhardt** [reply]
Thanks for the update @ghandlin. I don't suppose you've missed a `seq service restart` after applying the new settings? It seems like opening the firewall port, creating the NSG/security rule, and applying it to the VM's subnet, should do the job, otherwise 🤔

**Reply by ghandlin** [reply]
Nope, restarted after each config change. I'll look at it more today.
